src/ast/visitStmt.zig

```diff
@@ -1017,20 +1020,87 @@ pub fn VisitStmt(
+
+                        if (p.options.features.minify_syntax) {
+                            current_if_data.test_ = SideEffects.simplifyBoolean(p, current_if_data.test_);
+                        }
+
+                        const chain_effects = SideEffects.toBoolean(p, current_if_data.test_.data);
+                        if (chain_effects.ok and !chain_effects.value) {
+                            const old = p.is_control_flow_dead;
+                            p.is_control_flow_dead = true;
+                            current_if_data.yes = p.visitSingleStmt(current_if_data.yes, StmtsKind.none);
+                            p.is_control_flow_dead = old;
+                        } else {
+                            current_if_data.yes = p.visitSingleStmt(current_if_data.yes, StmtsKind.none);
+                        }
+
+                        // Move to the next else clause
+                        if (current_if_data.no) |next_no| {
+                            // Trim unnecessary "else" clauses
+                            if (p.options.features.minify_syntax) {
+                                if (@as(Stmt.Tag, next_no.data) == .s_empty) {
+                                    current_if_data.no = null;
+                                    break;
+                                }
+                            }
+
+                            // Check if the next else is another if (continuing the chain)
+                            if (next_no.data == .s_if) {
+                                current_no = next_no;
+                                parent_data = current_if_data;
+                                continue; // Continue processing the chain
+                            } else {
+                                // Not an else-if, process the final else branch normally
+                                if (chain_effects.ok and chain_effects.value) {
+                                    const old = p.is_control_flow_dead;
+                                    p.is_control_flow_dead = true;
+                                    defer p.is_control_flow_dead = old;
+                                    current_if_data.no = p.visitSingleStmt(next_no, .none);
+                                } else {
+                                    current_if_data.no = p.visitSingleStmt(next_no, .none);
+                                }
+
+                                // Trim unnecessary "else" clauses
+                                if (p.options.features.minify_syntax) {
+                                    if (current_if_data.no != null and @as(Stmt.Tag, current_if_data.no.?.data) == .s_empty) {
+                                        current_if_data.no = null;
+                                    }
+                                }
+                                break;
+                            }
+                        } else {
+                            break; // No more else clauses
+                        }
```

Comment:
Don't nest code like this. Do `const next_no = current_if_data.no orelse break;` instead.